### ISSUE TITLE
Go-like syntax to handle errors

### ISSUE THREAD

**Issue by enkot** [unjs/ofetch#370]
### Describe the feature

I suggest to add `throw: boolean` option so we can return the error instead of throwing it. This way users can get full type safety after implementation of [Typed API definition](https://github.com/unjs/ofetch/issues/364).
Example:
```ts
const { data, error } = $fetch('/api/v1', { throw: false })
```
or not throw by default:
```ts
const $api = $fetch.create({
  baseURL: 'http://httpstat.us',
  throw: false
})

const { data, error } = $fetch<unknown, ApiDefinition>('/500')

error // -> error is typed here based on the ApiDefinition
```

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by pi0** [comment]
I like the idea. Only thinking it might be more suitable for `$fetch.raw` where we return full object (data body, headers and now possibly new `headers`). Combining two different return values for `$fetch` can increase the types complexity at very least but open to ideas.

**Comment by anuragkumar19** [comment]
I fully agree with @pi0. 

@enkot, Generally APIs `ErrorResponse` schema is same for every routes. So, another way to  achieve it would be we expose ErrorResponse interface which will be used by `raw` to override the response type with `ErrorResponse` if success is false. `ErrorResponse` can overwritten via `.d.ts` file.

Currently the work around will be to let your api return `status_code:number` or `success:boolean` & make response a union and use it with `ignoreResponseError` to get correct typing.

**Comment by johannschopplich** [comment]
By the way, you could support both syntaxes in the same return object:

```ts
const { data, error } = $fetch('/api/v1', { throw: false })
// And:
const [data, error] = $fetch('/api/v1', { throw: false })
```

Implementation:

```ts
export function createIsomorphicDestructurable<
  T extends Record<string, unknown>,
  A extends readonly any[],
>(obj: T, arr: A): T & A {
  const clone = { ...obj }

  Object.defineProperty(clone, Symbol.iterator, {
    enumerable: false,
    value() {
      let index = 0
      return {
        next: () => ({
          value: arr[index++],
          done: index > arr.length,
        }),
      }
    },
  })

  return clone as T & A
}
```

[Source](https://antfu.me/posts/destructuring-with-object-or-array)

**Comment by enkot** [comment]
Good point 👍 
One more reference: https://vueuse.org/shared/makeDestructurable/#makedestructurable

**Comment by johannschopplich** [comment]
@enkot Absolutely! That's the same implementation under the hood. The VueUse one just also has IE11 support, probably because VueUse supports Vue 2 still. :)

**Comment by enkot** [comment]
> @enkot, Generally APIs `ErrorResponse` schema is same for every routes. So, another way to achieve it would be we expose ErrorResponse interface which will be used by `raw` to override the response type with `ErrorResponse` if success is false. `ErrorResponse` can overwritten via `.d.ts` file.

Agree, generally the same for all routes, but not 100%:
- there are situations when, for example, the project uses gateways (aggregators) to different APIs and error response can be quite different.
- an error response is just a different type of response (not 200), so why should it be treated differently in terms of types?

**Comment by letoast** [comment]
@enkot 
Was this implemented with 
``` js
await ofetch("/url", { ignoreResponseError: true });
```

https://github.com/unjs/ofetch?tab=readme-ov-file#%EF%B8%8F-handling-errors ?

**Comment by Winter979** [comment]
The only issue i have with ignoreResponseError is that it doesnt run the onRequestError hooks
